### PR TITLE
change to gzstream library

### PR DESCRIPTION
**Pull request by Disservin**
First PR for #76

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,9 +18,9 @@ ifeq ($(uname_S), Darwin)
 	NATIVE =	
 endif
 
-SRC_FILE = scoreWDLstat.cpp
+SRC_FILE = scoreWDLstat.cpp external/gzip/gzstream.cpp
```
Comment (vondele) [inline Makefile:21]: in this way, our formatting will format this file as well, which is probably not ideal for externals.
What about introducing an
EXT_SRC_FILE = external/gzip/gzstream.cpp
and adjusting the $(EXE_FILE) target (dependencies and compile part).

File: Makefile [hunk 2]
```diff
@@ -18,13 +18,14 @@ ifeq ($(uname_S), Darwin)
...
+SRC_FILE = scoreWDLstat.cpp
+EXT_SRC_FILE = external/gzip/gzstream.cpp
 EXE_FILE = scoreWDLstat
 HEADERS = external/chess.hpp external/json.hpp external/threadpool.hpp scoreWDLstat.hpp external/gzip/gzstream.h
 
 all: $(EXE_FILE)
 
-$(EXE_FILE): $(SRC_FILE) $(HEADERS)
+$(EXE_FILE): $(SRC_FILE) $(HEADERS) $(EXT_SRC_FILE)
 	$(CXX) $(CXXFLAGS) $(NATIVE) -o $(EXE_FILE) $(SRC_FILE) -lz
```
Comment (vondele) [inline Makefile:29]: missing $(EXT_SRC_FILE) on the CXX line ?

Comment (Disservin) [reply Makefile:29]: whops

**Comment by Disservin** [comment]
i'll rebase

**Comment by Disservin** [comment]
ah shit

**Comment by Disservin** [comment]
I've also made the header split into external and "internal"

**Pull request merged**